zkevm-circuits/src/tx_circuit.rs

```diff
@@ -1690,6 +1731,156 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
+                meta.query_advice(chunk_txbytes_len_acc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_len_acc, Rotation(-(HASH_RLC_OFFSET as i32)))
+                        // the previous row in fixed tx_table is the signed RLP length of current tx
+                        + meta.query_advice(tx_table.value, Rotation::prev()),
+            );
+
+            // Accumulate chunk bytes RLC
+            cb.require_equal(
+                "chunk_txbytes_rlc::cur == chunk_txbytes_rlc::prev * pow_of_rand(HashLength) + HashRLC",
+                meta.query_advice(chunk_txbytes_rlc, Rotation::cur()),
+                meta.query_advice(chunk_txbytes_rlc, Rotation(-(HASH_RLC_OFFSET as i32)))
+                        * meta.query_advice(pow_of_rand, Rotation::cur())
+                        + meta.query_advice(tx_table.value, Rotation::cur()),
+            );
+
+            // The chunk bytes len is the same as the HashLen field in tx_table (in the prev row)
+            cb.require_equal(
+                "chunk_bytes_len = HashLen",
+                meta.query_advice(chunk_bytes_len, Rotation::cur()),
+                meta.query_advice(tx_table.value, Rotation::prev()),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enable, Rotation::cur()),
+                // Only l2 signed bytes are accumulated
+                meta.query_advice(is_chunk_bytes, Rotation::cur()),
+                is_hash_rlc(meta),
+            ]))
+        });
+
+        meta.create_gate(
+            "Chunk Bytes RLC stays same for l1 msg and padding txs",
+            |meta| {
+                let mut cb = BaseConstraintBuilder::default();
+
+                // Check hash length is unchanged
+                cb.require_equal(
+                    "chunk_txbytes_len_acc::cur == chunk_txbytes_len_acc::prev",
+                    meta.query_advice(chunk_txbytes_len_acc, Rotation::cur()),
+                    meta.query_advice(chunk_txbytes_len_acc, Rotation(-(HASH_RLC_OFFSET as i32))),
+                );
+
+                // Check chunk RLC is unchanged
+                cb.require_equal(
+                    "chunk_txbytes_rlc::cur == chunk_txbytes_rlc::prev",
+                    meta.query_advice(chunk_txbytes_rlc, Rotation::cur()),
+                    meta.query_advice(chunk_txbytes_rlc, Rotation(-(HASH_RLC_OFFSET as i32))),
+                );
+
+                cb.gate(and::expr([
+                    meta.query_fixed(q_enable, Rotation::cur()),
+                    not::expr(meta.query_advice(is_chunk_bytes, Rotation::cur())),
+                    is_hash_rlc(meta),
+                ]))
+            },
+        );
+
+        meta.lookup_any("Correct pow_of_rand for HashLen", |meta| {
+            let enable = and::expr(vec![
+                meta.query_fixed(q_enable, Rotation::cur()),
+                not::expr(meta.query_advice(is_l1_msg, Rotation::cur())),
```

Comment:
The condition should be `q_enable && is_chunk_bytes && is_hash_rlc` as we use `pow_of_rand::cur` to accumulate `chunk_txbytes_rlc` at `HASH_RLC` row.